provisioning_templates/snippet/kickstart_ifcfg_generic_interface.erb

```diff
@@ -45,6 +53,7 @@ DEFROUTE=<%= primary %>
 <%- end -%>
 <%- if @interface.virtual? && (!@subnet.nil? && (@subnet.has_vlanid? || @interface.vlanid.present?)) -%>
 <%=   "VLAN=yes" %>
+<%=   "VLAN_ID=#{@interface.vlanid}" %>
```

Comment:
Currently the ID is taken from NIC identifier. What is your motivation for this change? This one seems ok.

Reply:
This is the result from using `nmcli` to get it all to work. The `nmcli` tool adds `VLAN_ID` explicitly.

See [the documentation](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/networking_guide/sec-configure_802_1q_vlan_tagging_using_the_command_line_tool_nmcli) on using `nmcli` to add a tagged vlan. Also [these lines](https://github.com/NetworkManager/NetworkManager/blob/c2f8400e6606918873505280d0edff3577e7d7ab/src/settings/plugins/ifcfg-rh/nms-ifcfg-rh-reader.c#L5520) in the source code. Device name still takes precedence if it matches.